tests/lib/types/types.test.ts

```diff
@@ -634,6 +634,17 @@ rule = {
     },
 };
 
+type DeprecatedRuleContextKeys = "getAncestors" | "getDeclaredVariables" | "getScope" | "markVariableAsUsed";
+(): RuleDefinition => ({
+    create(context) {
+        // Ensure that deprecated RuleContext methods are not defined when using RuleDefinition
+        context satisfies {
+            [Key in keyof typeof context]: Key extends DeprecatedRuleContextKeys ? never : typeof context[Key];
+        };
```

Comment:
Is there any minimum typescript version that we support? `satisies` was introduced in [v4.9.0](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html)

Reply:
We don't have a policy for that 🤷‍♂️ Currently, we are only running our type tests on the latest version of TypeScript.